src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatter.cs

```diff
@@ -114,15 +116,18 @@ public virtual Encoding SelectCharacterEncoding(OutputFormatterWriteContext cont
                 return encoding;
             }
 
-            var charset = context.ContentType?.Charset;
-            if (charset != null)
+            if (context.ContentType.HasValue)
             {
-                for (var i = 0; i < SupportedEncodings.Count; i++)
+                var contentTypeEncoding = MediaTypeEncoding.GetEncoding(context.ContentType);
```

Comment:
Since `SupportedEncodings` is a thing... would it be better to just return the charset here?